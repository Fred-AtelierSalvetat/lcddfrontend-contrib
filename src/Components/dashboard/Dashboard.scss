$navbarwidth: 80px;
$navbartopmargin: 46px;
$cardheight: calc(1.875rem + 1.5rem + 2.375rem); // SVG + 3 * padding + 2 txt lines
$cardwidth: calc(#{$navbarwidth} - 2px);

#lcdd-dashboard {
    p {
        margin: 0;
        margin-block-start: 0;
        margin-block-end: 0;
    }

    display: flex;
    height: calc(100vh - 72px);

    #lcdd-dashboard-navbar {
        position: relative;
        z-index: 1;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: flex-start;
        min-width: $navbarwidth;
        max-width: $navbarwidth;
        width: $navbarwidth;
        height: 100%;

        overflow-y: auto;
        overflow-x: hidden;
        padding: $navbartopmargin 1px 0 1px;
        background-color: var(--dark-primary);

        > input {
            display: none;
        }
        > label {
            display: flex;
            align-items: center;
            justify-content: center;
            min-width: $cardwidth;
            max-width: $cardwidth;
            width: $cardwidth;
            height: $cardheight;

            cursor: pointer;
            margin: 0.5rem 0;

            transition: all 0.2s ease-in-out;

            &:hover {
                transform: scale(1.1);
            }

            p {
                font-weight: 400;
                font-size: small;
                font-size: 13px;
                line-height: 19px;
                color: var(--white);
                text-align: center;
                pointer-events: none;
            }
            svg {
                width: 1.875rem;
                height: 1.875rem;
                fill: var(--white);
                padding-bottom: 0.5rem;
            }
        }
        > input:checked + label {
            color: var(--dark-primary);
            cursor: default;
            &:hover {
                transform: none;
            }
            p {
                color: var(--dark-primary);
            }
            svg {
                fill: var(--dark-primary);
            }
        }

        .selection-sliding-background {
            display: none;
            min-width: $cardwidth;
            max-width: $cardwidth;
            width: $cardwidth;
            height: $cardheight;
            min-height: $cardheight;
            max-height: $cardheight;
            transition: top 0.2s ease-in-out;
            border-radius: 2px;
        }
        input:checked ~ .selection-sliding-background {
            display: block;
            position: absolute;
            z-index: -1;

            background-color: var(--white);
            top: 0;
        }
        input:first-of-type:checked ~ .selection-sliding-background {
            top: calc(#{$navbartopmargin} + 0.5rem);
        }
        input:nth-of-type(2):checked ~ .selection-sliding-background {
            top: calc(#{$navbartopmargin} + 0.5rem + (#{$cardheight} + 1rem));
        }
        input:nth-of-type(3):checked ~ .selection-sliding-background {
            top: calc(#{$navbartopmargin} + 0.5rem + 2 * (#{$cardheight} + 1rem));
        }
        input:nth-of-type(4):checked ~ .selection-sliding-background {
            top: calc(#{$navbartopmargin} + 0.5rem + 3 * (#{$cardheight} + 1rem));
        }
        input:nth-of-type(5):checked ~ .selection-sliding-background {
            top: calc(#{$navbartopmargin} + 0.5rem + 3 * (#{$cardheight} + 1rem));
        }
        input:nth-of-type(6):checked ~ .selection-sliding-background {
            top: calc(#{$navbartopmargin} + 0.5rem + 3 * (#{$cardheight} + 1rem));
        }
    }

    #lcdd-dashboard-page {
        height: 100%;
        width: calc(100% - #{$navbarwidth});
        padding: 0 36px;
        background: linear-gradient(180deg, #ffffff 0%, #ffffff 23.96%, #f2f2f2 100%);
        overflow: auto;
    }
}
