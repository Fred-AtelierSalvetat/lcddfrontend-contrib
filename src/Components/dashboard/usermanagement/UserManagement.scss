$line-separator-color: #eceeef;
$headers-color: #eceeef;
$header-background-color: #f2f2f2;
$input-border-color: #cccccc;
$input-place-holder-color: #999999;
$awaiting-validation-background-color: #f0ad4e;

#lcdd-dashboard-page {
    .table_and_filters_pane {
        background-color: $header-background-color;
        border-top-left-radius: 0.25rem;
        border-top-right-radius: 0.25rem;
        //width: fit-content;
        min-width: 100%;
        filter: drop-shadow(0 0.25rem 0.5rem rgba(130, 130, 130, 0.5));
        .nav {
            padding: 0.5rem;

            .nav-link {
                text-decoration: none;
                transition: transform 0.2s;
                padding: 0.5rem;
                color: var(--primary);
                background-color: transparent;
                &:hover {
                    transform: scale(1.1);
                }
                &.nav-link.active {
                    color: var(--white);
                    background-color: var(--primary);
                    &:hover {
                        transform: none;
                    }
                }
            }
            .searchbox-container {
                width: calc(100vw - 160px);
                position: relative;
                align-self: bottom;

                @media (min-width: 769px) {
                    margin-left: auto;
                    min-width: 200px;
                    width: 400px;
                    align-self: top;
                }
            }
        }

        .tab-content {
            background-color: var(--white);
            margin-bottom: 7px;
            .tab-pane {
                height: calc(100vh - 200px);
                min-height: 97px;
                overflow: auto;
            }
            thead {
                tr {
                    th {
                        position: sticky;
                        top: 0;
                        z-index: 10;
                        background-color: var(--gray-dark);
                        color: $headers-color;
                        padding: 0.75rem 1rem;
                        font-style: normal;
                        font-weight: 700;
                        line-height: 1, 5rem;
                        letter-spacing: 0;
                        text-align: left;
                        vertical-align: middle;
                        white-space: nowrap;
                        > .CenteredHeader {
                            text-align: center;
                        }
                    }
                }
            }
            tbody {
                tr {
                    border-top: 1px solid $line-separator-color;
                    td {
                        z-index: 9;
                        padding: 0.75rem 1rem;
                        background-color: var(--white);
                        .Centered {
                            display: flex;
                            align-items: center;
                            justify-content: center;

                            .speakerStatus {
                                color: var(--white);
                                border-radius: 20px;

                                height: 20px;
                                width: 68px;
                                display: flex;
                                align-items: center;
                                justify-content: center;
                                margin: 4px 0 6px 0;
                                &.speakerAwaitingAnswer {
                                    background-color: var(--cyan);
                                }
                                &.speakerAwaitingValidation {
                                    background-color: $awaiting-validation-background-color;
                                }
                                p {
                                    margin: 0;
                                    line-height: 22px;
                                    font-size: 12px;
                                    font-style: normal;
                                    font-weight: 700;
                                    letter-spacing: 0px;
                                    white-space: nowrap;
                                }
                            }
                        }

                        > p {
                            margin: 0;
                            font-style: normal;
                            font-weight: 400;
                            line-height: 1.5rem;
                            letter-spacing: 0px;
                            text-align: left;
                            vertical-align: middle;
                            white-space: nowrap;
                        }
                        > a {
                            font-size: small;
                            font-weight: 400;
                            font-style: normal;
                            white-space: nowrap;
                            color: var(--link);
                        }
                        .userStatus {
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            > svg {
                                cursor: pointer;

                                &:hover {
                                    transform: scale(1.1);
                                }
                                #toggle_on_24px {
                                    fill: var(--success);
                                }
                                #toggle_off_24px {
                                    fill: var(--danger);
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
