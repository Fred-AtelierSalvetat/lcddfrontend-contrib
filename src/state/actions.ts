import { QuestionsActions } from './questions/questions.actions';
import { SpeakersActions } from './speaker/speaker.action';

export type LcddActions = QuestionsActions | SpeakersActions;
