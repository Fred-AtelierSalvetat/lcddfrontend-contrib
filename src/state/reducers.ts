import { combineReducers } from 'redux';
import { homeReducer } from './questions/questions.reducer';
import { SpeakerReducer } from './speaker/speaker.reducer';
import users from './users/reducer';
import alerts from './alerts/reducer';
import type { Alert } from './alerts/model';

import * as fromUsers from './users/selectors';
import * as fromAlerts from './alerts/selectors';
import * as fromUser from './user/selectors';
import { roleFilterMap } from './users/constants/roles';

import workshops from './workshops/reducer';
import type { WorkshopsState, Workshop, SearchFilter } from './workshops/model';
import * as fromWorkshops from './workshops/selectors';

import { userRegistrationReducer } from './user/user.registation.reducer';
import { userAuthenticationReducer } from './user/user.authentication.reducer';

import type { RootStateType } from './store';
import type { UsersActionType } from './users/constants/actionTypes';
import type { User, UIfiltersRole, UIfiltersSearch } from './users/model';

export const reducersList = {
    home: homeReducer,
    speakers: SpeakerReducer,
    users,
    workshops,
    alerts,
    registration: userRegistrationReducer,
    authentication: userAuthenticationReducer,
};

export const rootReducer = combineReducers(reducersList);

export type AppState = ReturnType<typeof rootReducer>;

export const getRoles: () => UIfiltersRole[] = () => Object.keys(roleFilterMap);
export const roleFilterSelector = (state: RootStateType): UIfiltersRoles => fromUsers.roleFilterSelector(state.users);
export const searchFilterSelector = (state: RootStateType): UIfiltersSearch =>
    fromUsers.searchFilterSelector(state.users);
export const getVisibleUsers = (state: RootStateType): User[] => fromUsers.getVisibleUsers(state.users);
export const isRequestInProgress =
    (request_type: UsersActionType) =>
    (state: RootStateType): boolean =>
        fromUsers.isRequestInProgress(state.users, request_type);
export const getAlerts = (state: RootStateType): Alert[] => fromAlerts.getAlerts(state.alerts);
export const isCurrentUserLoggedIn = (state: RootStateType): boolean =>
    fromUser.isCurrentUserLoggedIn(state.authentication);
export const getWorkshops = (state: WorkshopsState): Workshop[] => fromWorkshops.getWorkshops(state.workshops);
export const workshopSearchFilterSelector = (state: WorkshopsState): SearchFilter =>
    fromWorkshops.searchFilterSelector(state.workshops);
export const getWorkshopById =
    (id: Workshop.id) =>
    (state: RootStateType): Workshop =>
        fromWorkshops.getWorkshopById(state.workshops, id);
export const isWorkshopStoreInialized = (state: RootStateType): boolean =>
    fromWorkshops.isWorkshopStoreInialized(state.workshops);
export const getOrderBy = (state: RootStateType): Workshop.orderBy => fromWorkshops.getOrderBy(state.workshops);
