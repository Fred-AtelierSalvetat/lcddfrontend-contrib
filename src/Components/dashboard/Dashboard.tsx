import React, { Fragment, FC } from 'react';
import { useLocation, useHistory, matchPath, Redirect, Switch, Route } from 'react-router-dom';
import { ReactComponent as WorkShopsIcon } from '~/assets/icons/workshops_24px.svg';
import { ReactComponent as WorkshopCreationIcon } from '~/assets/icons/add_24px.svg';
// import { ReactComponent as QuestionsIcon } from '~/assets/icons/questions_24px.svg';
import { ReactComponent as UsersIcon } from '~/assets/icons/users_24px.svg';
// import { ReactComponent as AnalyticsIcon } from '~/assets/icons/analytics_24px.svg';
// import { ReactComponent as NotificationsIcon } from '~/assets/icons/notifications_24px.svg';
// import { ReactComponent as HelpIcon } from '~/assets/icons/help_24px.svg';
import ErrorBoundary from '~/Components/shared/ErrorBoundary';

import AlertNotificationBox from './AlertNotificationBox';

import './Dashboard.scss';

const UserManagement = React.lazy(() => import('./usermanagement/UserManagement'));
const NewWorkshop = React.lazy(() => import('./newWorkshop/NewWorkshop'));
const Workshops = React.lazy(() => import('./workshops/Workshops'));
const EditWorkshop = React.lazy(() => import('./editWorkshop/EditWorkshop'));
const GoLive = React.lazy(() => import('./goLive/GoLive'));

const Dashboard: FC = () => {
    const location = useLocation();
    const history = useHistory();

    interface PageDesc {
        icon: JSX.Element;
        label: JSX.Element;
        href: string;
        page: JSX.Element;
    }
    const pages: PageDesc[] = [
        {
            icon: <WorkShopsIcon />,
            label: <p>Ateliers</p>,
            href: '/dashboard/workshops',
            page: <Workshops />,
        },
        // {
        //     icon: <QuestionsIcon />,
        //     label: <p>Questions</p>,
        //     href: '#/dashboard/questions',
        //     page: <p>Questions</p>,
        // },
        {
            icon: <WorkshopCreationIcon />,
            label: (
                <p>
                    Cr√©er
                    <br />
                    un atelier
                </p>
            ),
            href: '/dashboard/newWorkshop',
            page: <NewWorkshop />,
        },
        {
            icon: <UsersIcon />,
            label: (
                <p>
                    Gestion
                    <br />
                    utilisateurs
                </p>
            ),
            href: '/dashboard/users',
            page: <UserManagement />,
        },
        {
            href: '/dashboard/editWorkshop/:id',
            page: <EditWorkshop />,
        },
        {
            href: '/dashboard/goLive/:id',
            page: <GoLive />,
        },
        // {
        //     icon: <AnalyticsIcon />,
        //     label: <p>Analytics</p>,
        //     href: '#/dashboard/analytics',
        //     page: <p>Analytics</p>,
        //     disabled: true,
        // },
        // {
        //     icon: <NotificationsIcon />,
        //     label:  <p>Notifications</p>,
        //     href: '#/dashboard/notifications',
        //     page: <p>Notifications</p>,
        //     disabled: true,
        // },
        // {
        //     icon: <HelpIcon />,
        //     label: <p>Aide</p>,
        //     href: '#/dashboard/help',
        //     page: <p>Help</p>,
        //     disabled: true,
        // },
    ];

    return (
        <ErrorBoundary>
            <div id="lcdd-dashboard">
                <div id="lcdd-dashboard-navbar">
                    {pages
                        .filter((page) => !!page.icon)
                        .map((page_desc) => (
                            <Fragment key={page_desc.href}>
                                <input
                                    type="radio"
                                    id={page_desc.href}
                                    value={page_desc.href}
                                    checked={
                                        !!matchPath(location.pathname, {
                                            path: page_desc.href,
                                            exact: true,
                                            strict: true,
                                        })
                                    }
                                    onChange={(changeEvent) => history.push(changeEvent.target.value)}
                                />
                                <label htmlFor={page_desc.href}>
                                    <div>
                                        {page_desc.icon}
                                        {page_desc.label}
                                    </div>
                                </label>
                            </Fragment>
                        ))}
                    <div className="selection-sliding-background" />
                </div>
                <div id="lcdd-dashboard-page">
                    <AlertNotificationBox />
                    <Switch>
                        {pages.map((page_desc) => (
                            <Route exact strict key={page_desc.href} path={page_desc.href}>
                                {page_desc.page}
                            </Route>
                        ))}
                        <Route exact strict path="/dashboard" />
                        <Route path="*">
                            <Redirect to="/no-match" />;
                        </Route>
                    </Switch>
                </div>
            </div>
        </ErrorBoundary>
    );
};

export default Dashboard;
