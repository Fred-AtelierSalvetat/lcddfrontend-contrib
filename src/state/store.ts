import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { setupListeners } from '@reduxjs/toolkit/query/react';
import api from '../api/lcddbackend-api';
import { reducersList } from './reducers';

export const store = configureStore({
    reducer: { ...reducersList, [api.reducerPath]: api.reducer },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(api.middleware, logger),
    devTools: process.env.NODE_ENV !== 'production',
});
setupListeners(store.dispatch);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootStateType = ReturnType<typeof store.getState>;
export type AppDispatchType = typeof store.dispatch;
