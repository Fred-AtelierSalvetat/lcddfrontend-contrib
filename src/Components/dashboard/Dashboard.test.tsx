import React, { Suspense } from 'react';
import { expect } from 'chai';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { MemoryRouter, Route } from 'react-router-dom';

import Dashboard from './Dashboard';
import UserManagement from './usermanagement/UserManagement';

import { rootReducer, reducersList } from '~/state/reducers';

jest.mock('./usermanagement/UserManagement');

describe('<Dashboard />', () => {
  let store;
  beforeEach(() => {
    const mockStore = configureMockStore([thunk]);
    const undefinedGlobalState = {};
    for (const prop of Object.getOwnPropertyNames(reducersList)) {
      undefinedGlobalState[prop] = undefined;
    }
    store = mockStore(rootReducer(undefinedGlobalState, { type: undefined }));

    UserManagement.mockImplementation(() => <div id="TEST-TARGET">{testPage}</div>);
  });

  it('contains 3 navigation links', () => {
    const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/dashboard']}>
                    <Route path="/dashboard" exact>
                        <Dashboard />
                    </Route>
                </MemoryRouter>
                ,
            </Provider>,
    );
    expect(wrapper.find('input')).to.have.length(3);
  });
  it('contains a link to /dashboard/newWorkshop', () => {
    const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/dashboard']}>
                    <Route path="/dashboard" exact>
                        <Dashboard />
                    </Route>
                </MemoryRouter>
                ,
            </Provider>,
    );
    expect(wrapper.find('input').find({ value: '/dashboard/newWorkshop' })).to.exist;
  });
  it('contains a link to /dashboard/users', () => {
    const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/dashboard']}>
                    <Route path="/dashboard" exact>
                        <Dashboard />
                    </Route>
                </MemoryRouter>
            </Provider>,
    );
    expect(wrapper.find('input').find({ value: '/dashboard/users' })).to.exist;
  });
  it('contains a link to /dashboard/newWorkshop', () => {
    const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/dashboard']}>
                    <Route path="/dashboard" exact>
                        <Dashboard />
                    </Route>
                </MemoryRouter>
            </Provider>,
    );
    expect(wrapper.find('input').find({ value: '/dashboard/newWorkshop' })).to.exist;
  });
  it('contains a link to /dashboard/workshops', () => {
    const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/dashboard']}>
                    <Route path="/dashboard" exact>
                        <Dashboard />
                    </Route>
                </MemoryRouter>
            </Provider>,
    );
    expect(wrapper.find('input').find({ value: '/dashboard/workshops' })).to.exist;
  });

  it("shall use route's location to activate the corresponding 'input' and display contents", () => {
    const testPage = 'UserManagementTESTPAGE';
    const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/dashboard/users']}>
                    <Route path="/dashboard/users" exact>
                        <Suspense fallback={<p>"Loading"</p>}>
                            <Dashboard />
                        </Suspense>
                    </Route>
                </MemoryRouter>
                ,
            </Provider>,
    );
    console.log('WRAPPER =', wrapper.debug());
    expect(wrapper.find('input').filter({ value: '/dashboard/users' })).to.have.length(1);
    expect(wrapper.find('input').filter({ value: '/dashboard/users' }).at(0).props().checked).to.equal(true);

    expect(wrapper.find('#TEST-TARGET').childAt(0)).to.contain(testPage);
  });

  it('should match its reference snapshot', () => {
    const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={[{ pathname: '/dashboard', key: 'testKey' }]}>
                    <Route path="/dashboard" exact>
                        <Dashboard />
                    </Route>
                </MemoryRouter>
                ,
            </Provider>,
    );
    expect(wrapper).to.matchSnapshot();
  });
});
