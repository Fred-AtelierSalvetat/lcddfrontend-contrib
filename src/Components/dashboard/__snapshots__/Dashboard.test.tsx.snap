// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should match its reference snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/dashboard",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/dashboard",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/dashboard",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Route
        exact={true}
        path="/dashboard"
      >
        <Dashboard>
          <ErrorBoundary>
            <div
              id="lcdd-dashboard"
            >
              <div
                id="lcdd-dashboard-navbar"
              >
                <input
                  checked={false}
                  id="/dashboard/workshops"
                  onChange={[Function]}
                  type="radio"
                  value="/dashboard/workshops"
                />
                <label
                  htmlFor="/dashboard/workshops"
                >
                  <div>
                    <ForwardRef(SvgWorkshops24Px)>
                      <svg>
                        workshops_24px.svg
                      </svg>
                    </ForwardRef(SvgWorkshops24Px)>
                    <p>
                      Ateliers
                    </p>
                  </div>
                </label>
                <input
                  checked={false}
                  id="/dashboard/newWorkshop"
                  onChange={[Function]}
                  type="radio"
                  value="/dashboard/newWorkshop"
                />
                <label
                  htmlFor="/dashboard/newWorkshop"
                >
                  <div>
                    <ForwardRef(SvgAdd24Px)>
                      <svg>
                        add_24px.svg
                      </svg>
                    </ForwardRef(SvgAdd24Px)>
                    <p>
                      Cr√©er
                      <br />
                      un atelier
                    </p>
                  </div>
                </label>
                <input
                  checked={false}
                  id="/dashboard/users"
                  onChange={[Function]}
                  type="radio"
                  value="/dashboard/users"
                />
                <label
                  htmlFor="/dashboard/users"
                >
                  <div>
                    <ForwardRef(SvgUsers24Px)>
                      <svg>
                        users_24px.svg
                      </svg>
                    </ForwardRef(SvgUsers24Px)>
                    <p>
                      Gestion
                      <br />
                      utilisateurs
                    </p>
                  </div>
                </label>
                <div
                  className="selection-sliding-background"
                />
              </div>
              <div
                id="lcdd-dashboard-page"
              >
                <AlertNotificationBox>
                  <div
                    className="alertBox"
                  />
                </AlertNotificationBox>
                <Switch>
                  <Route
                    computedMatch={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/dashboard",
                        "url": "/dashboard",
                      }
                    }
                    exact={true}
                    location={
                      Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/dashboard",
                        "search": "",
                      }
                    }
                    path="/dashboard"
                    strict={true}
                  />
                </Switch>
              </div>
            </div>
          </ErrorBoundary>
        </Dashboard>
      </Route>
    </Router>
  </MemoryRouter>
  ,
</Provider>
`;
